### PR TITLE
Displaying summary

### PR DESCRIPTION
**Pull request by flbulgarelli**
### :dart: PR Goal

This PR add supports for summaries: short text messages generated by the runner that try to capture the essence of the test-failure, hiding lower-level details - which continue con live in the results.

Such summaries are important for the following reasons: 

 - They may be generated in the user's language instead of just plain english
 - They become first citizens in the platform instead of being tangled within some runners code
 - They come also with a symbolic key, which allows to perform better analysis on data
 - They are displayable in kids layout :confetti_ball: 

### :eyes: Relations

 * Related to #1364 
 * Related to #1397 

### :ballot_box_with_check: Pending tasks

- [x] Fix tests
- ~Change bubble text when there switching in multiple scenarios~
- [x] Do more integration testing
- ~Support markdown~ I will support in the next #1400 PR, since it is a little more complex to do this in a appropriate manner

### :spiral_notepad:  More details

[Single feedback message proposal](https://docs.google.com/document/d/1-wFSTVtjwEM4hieVb7I-AAeXngmx_Sx77AeEQW-P1Bk/edit#)

##### Standard layout

summary_1

##### Standard layout with gobstones multi-scenario exercise with title tests and summary

image

##### Standard layout with gobstones multi-scenario exercise with untitled tests

image
#### Standard layout with gobstones single-scenario exercise with untitled test

image
 

#### Kids layout, different head 

image

##### Kids layout, different board

image

##### Kids layout, when no summary present 

:warning:  _This is what will happen if gobstones runner does not yet support summary_

image

###### Standard layout, when summary not present in gobstones exercise

:warning:  _This is what will happen if gobstones runner does not yet support summary_

image

### CODE REVIEW

**Comment by flbulgarelli** [comment]
Standard layout: 

summary_1

**Comment by flbulgarelli** [comment]
Kids layout: 

image

**Comment by flbulgarelli** [comment]
Some doubts: 

- Should the :x: be displayed before the summary message?
- Does it make sense to display that :x: on passes with warnings errors?
- It looks like there is nothing that prevents multiple expectation results to be displayed, right? They are just not visible. Isn't that wrong?
- Should we render that part client-side or server-side? I think we should do that client-side, in both passed_with_warnings and fail results. That may conflict with client-side rendering in this never-merged fork https://github.com/flbulgarelli/mumuki-laboratory/, but I think it is the right thing to do.

**Comment by flbulgarelli** [comment]
Pending tasks: 

- [ ] Fix tests
- [ ] Change bubble text when there switching in multiple scenarios 
- [ ] Do more integration testing

**Comment by flbulgarelli** [comment]
Standard layout with gobstones multi-scenario exercise: 

image

**Comment by flbulgarelli** [comment]
Standard layout with gobstones single-scenario exercise with untitled test

image

**Comment by flbulgarelli** [comment]
Some answers after discussion: 

> - Should the :x: be displayed before the summary message?

Nope

> Does it make sense to display that :x: on passes with warnings errors?

Nope

> Should we render that part client-side or server-side? I think we should do that client-side, in both passed_with_warnings and fail results. That may conflict with client-side rendering in this never-merged fork https://github.com/flbulgarelli/mumuki-laboratory/, but I think it is the right thing to do.

We should make rendering full client-side. I am going to tackle that on #1400

**Comment by flbulgarelli** [comment]
> Also there is a subtle inconsistency in colors in kids and standard layout: in kids test result - as well as expectation results - are shown in black font. But in standard layout, red is used. Should be use black in standard layout, too?

I will work on this in #1400

File: app/assets/javascripts/mumuki_laboratory/application/kids.js [hunk 1]
```diff
@@ -173,13 +173,17 @@ mumuki.load(function () {
     },
 
     _showMessageOnCharacterBubble: function (data) {
-      var $bubble = mumuki.kids._getCharacterBubble();
+      let $bubble = mumuki.kids._getCharacterBubble();
```
Comment (luchotc) [inline app/assets/javascripts/mumuki_laboratory/application/kids.js:176]: Prefer const over let if we aren't mutating them afterwards

**Comment by luchotc** [comment]
I like how it looks! Maybe we can rethink the whole layout, but I think that with the current one that's the best place to add it. That applies for both adults and kids layout

File: app/assets/javascripts/mumuki_laboratory/application/kids.js (hunk 1 above)
Comment (flbulgarelli) [reply app/assets/javascripts/mumuki_laboratory/application/kids.js:176]: I will fix that in #1400 , since it involves some big refactors

**Pull request merged**